tests/scripts/thread-cert/Cert_6_1_01_RouterAttach.py

```diff
@@ -51,19 +55,72 @@ def setUp(self):
         self.nodes[ED].add_whitelist(self.nodes[LEADER].get_addr64())
         self.nodes[ED].enable_whitelist()
 
+        self.sniffer = config.create_default_thread_sniffer(SNIFFER)
+        self.sniffer.start()
+
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
         self.nodes[LEADER].start()
-        self.nodes[LEADER].set_state('leader')
+        time.sleep(5)
         self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
 
         self.nodes[ED].start()
         time.sleep(5)
         self.assertEqual(self.nodes[ED].get_state(), 'child')
 
+        leader_messages = self.sniffer.get_messages_sent_by(LEADER)
+        ed_messages = self.sniffer.get_messages_sent_by(ED)      
+
+        # 1 - leader
+        msg = leader_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+
+        # 2 - ed
+        msg = ed_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
+        msg.assertSentWithHopLimit(255)
+        msg.assertSentToDestinationAddress("ff02::2")
+        msg.assertMleMessageContainsTlv(mle.Mode)
+        msg.assertMleMessageContainsTlv(mle.Challenge)
+        msg.assertMleMessageContainsTlv(mle.ScanMask)
+        msg.assertMleMessageContainsTlv(mle.Version)
+
+        scan_mask_tlv = msg.get_mle_message_tlv(mle.ScanMask)
+        self.assertEqual(1, scan_mask_tlv.router)
+        self.assertEqual(0, scan_mask_tlv.end_device)
+
+        self.assertEqual(0x02, msg.mle.aux_sec_hdr.key_id_mode)
+        
+        # 3 - leader
+        msg = leader_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
+        msg.assertSentToNode(self.nodes[ED])
+
+        # 4 - ed
+        msg = ed_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
+        msg.assertSentToNode(self.nodes[LEADER])
```

Comment:
`key identifier mode` should also be checked for Child Id Request (step 4 in Test Plan).